src/groups/mqb/mqbs/mqbs_virtualstoragecatalog.cpp

```diff
@@ -61,21 +61,21 @@ VirtualStorageCatalog::put(const bmqt::MessageGUID& msgGUID,
                            const mqbu::StorageKey&  appKey)
 {
     if (!appKey.isNull()) {
-        VirtualStoragesIter it = d_virtualStorages.find(appKey);
+        VirtualStoragesIter it = d_virtualStorages.findByKey2(appKey);
         BSLS_ASSERT_SAFE(it != d_virtualStorages.end());
 
-        return it->second->put(msgGUID,
-                               msgSize,
-                               rdaInfo,
-                               subScriptionId);  // RETURN
+        return it->value()->put(msgGUID,
+                                msgSize,
+                                rdaInfo,
+                                subScriptionId);  // RETURN
     }
 
     // Add guid to all virtual storages.
 
     for (VirtualStoragesIter it = d_virtualStorages.begin();
          it != d_virtualStorages.end();
          ++it) {
-        it->second->put(msgGUID, msgSize, rdaInfo, subScriptionId);
+        it->value()->put(msgGUID, msgSize, rdaInfo, subScriptionId);
```

Comment:
We don't check for return codes here, what if one or few `put` fails?
Currently, `VirtualStorage::put` might fail if GUID is not unique
On comparison, we might return `e_GUID_NOT_UNIQUE` on line 67

Reply:
Looks like this codepath (`appKey.isNull()`) is for unique GUIDs only.  The callers do not event check the return code.
We can add error code without affecting anything, I guess